src/client/components/Header/Header.component.js

```diff
@@ -21,19 +26,35 @@ export default function Header() {
       </Link>
       {/* ICONS */}
       <div className="icons-right">
-        <Link to="/shopping-cart"><button type='button' className='cart'><img className="icons" src={faShoppingCart} alt="shopping-cart" /></button></Link>
-        <img className="icons" src={faHeart} alt="favorite" />
+        <Link to="/shopping-cart">
+          <button className="icons cart" type="button">
+            <div style={{ position: 'relative' }}>
+              <img src={faShoppingCart} alt="shopping-cart" />
+              {numberOfItemsInCart >= 1 ? (
+                <span className="cart-item-number">{numberOfItemsInCart}</span>
+              ) : null}
+            </div>
+          </button>
+        </Link>
+        <div className="icons" style={{ position: 'relative' }}>
+          <img src={faHeart} alt="favorite" />
+          {numberOfItemsInFavorite >= 1 ? (
+            <span className="favorite-item-number">
+              {numberOfItemsInFavorite}
+            </span>
+          ) : null}
+        </div>
+
         <img className="icons" src={faUser} alt="login" />
 
```

Comment:
Authentication is not working properly, I can add items to cart even if I am not logged in.
The below code might help you and it is working fine with the function but need to style a bit.
Try replace the code from line 29 to 49.
````
{isAuthenticated ? (
          <div>
            <Link to="/shopping-cart">
              <button className="icons cart" type="button">
                <div style={{ position: 'relative' }}>
                  <img src={faShoppingCart} alt="shopping-cart" />
                  {numberOfItemsInCart >= 1 ? (
                    <span className="cart-item-number">
                      {numberOfItemsInCart}
                    </span>
                  ) : null}
                </div>
              </button>
            <img className="icons" src={faHeart} alt="favorite" />
            <img className="icons" src={faUser} alt="login" />
            </Link>
          </div>
        ) : (
          <div>
            <Link to="/shopping-cart">
              <button type="button" className="cart">
                <img
                  className="icons"
                  src={faShoppingCart}
                  alt="shopping-cart"
                />
              </button>
            </Link>
            <img className="icons" src={faHeart} alt="favorite" />
          </div>
        )}
````